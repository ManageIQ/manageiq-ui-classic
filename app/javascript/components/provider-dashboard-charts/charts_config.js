import moment from 'moment';

const dailyTimeTooltip = (units = undefined) => (data) => {
  const theMoment = moment(data[0].date);
  let htmlString = `<div class="tooltip-inner">${theMoment.format('L')}  ${data[0].value} ${data[0].group} </div>`;
  if (units) {
    htmlString = `<div class="tooltip-inner">${theMoment.format('L')}  ${data[0].value} ${__(units)} </div>`;
  }
  return htmlString;
};

const dailyPodTimeTooltip = () => (data) => {
  const theMoment = moment(data[0].date);

  if (data[1]) {
    return `<div class="tooltip-inner">
  ${theMoment.format('L')}  ${data[0].value} ${__(data[0].group)}, ${data[1].value} ${__(data[1].group)} </div>`;
  }
  return `<div class="tooltip-inner">${theMoment.format('L')}  ${data[0].value} ${__(data[0].group)} </div>`;
};

const hourlyPodTimeTooltip = () => (data) => {
  const theMoment = moment(data[0].date);
  if (data[1]) {
    return `<div class="tooltip-inner">
  ${theMoment.format('LT')}  ${data[0].value} ${__(data[0].group)}, ${data[1].value} ${__(data[1].group)} </div>`;
  }
  return `<div class="tooltip-inner">
  ${theMoment.format('LT')}  ${data[0].value} ${__(data[0].group)} </div>`;
};

const hourlyTimeTooltip = (units = undefined) => (data) => {
  const theMoment = moment(data[0].date);
  let htmlString = `<div class="tooltip-inner">${theMoment.format('LT')}  ${data[0].value} </div>`;
  if (units) {
    htmlString = `<div class="tooltip-inner">${theMoment.format('LT')}  ${data[0].value} ${__(units)} </div>`;
  }
  return htmlString;
};

export const chartConfig = {
  cpuUsageConfig: {
    chartId: 'cpuUsageChart',
    title: __('CPU'),
    units: __('Cores'),
    usageDataName: __('Used'),
    availableDataName: __('Available'),
    legendLeftText: __('Last 30 Days'),
    legendRightText: '',
    numDays: 30,
    availableof: __('Available of'),
    sparklineTooltip: () => dailyTimeTooltip(),
  },

  cpuUsageDonutConfig: {
    chartId: 'cpuDonutChart',
    thresholds: { warning: '60', error: '90' },
  },
  memoryUsageConfig: {
    chartId: 'memUsageChart',
    title: __('Memory'),
    units: __('GB'),
    usageDataName: __('Used'),
    availableDataName: __('Available'),
    legendLeftText: __('Last 30 Days'),
    legendRightText: '',
    numDays: 30,
    availableof: __('Available of'),
    sparklineTooltip: () => dailyTimeTooltip(),
  },

  memoryUsageDonutConfig: {
    chartId: 'memoryDonutChart',
    thresholds: { warning: '60', error: '90' },
  },
  recentResourcesConfig: {
    chartId: 'recentResourcesChart',
    tooltipFn: () => dailyTimeTooltip(),
    point: { r: 1 },
    size: { height: '145px' },
    grid: { y: { show: false } },
    setAreaChart: true,
    units: __('Hosts'),
    legendLeftText: __('Last 30 Days'),
  },
  recentVmsConfig: {
    chartId: 'recentVmsChart',
    headTitle: __('Recent VMs'),
    label: __('VMs'),
    tooltipFn: () => dailyTimeTooltip(),
    point: { r: 1 },
    size: { height: '145px' },
    grid: { y: { show: false } },
    setAreaChart: true,
    units: __('Vms'),
    legendLeftText: __('Last 30 Days'),
  },
  dailyNetworkUsageConfig: {
    chartId: 'networkUsageDailyChart',
    headTitle: __('Network Utilization Trend'),
    timeFrame: __('Last 30 Days'),
    units: __('KBps'),
    dataName: __('KBps'),
    tooltipFn: () => dailyTimeTooltip('KBps'),
    size: { height: '150px' },
    createdLabel: __('Network'),
    valueType: 'actual',
  },
  hourlyNetworkUsageConfig: {
    chartId: 'networkUsageHourlyChart',
    headTitle: __('Network Utilization Trend'),
    timeFrame: __('Last 24 Hours'),
    units: __('KBps'),
    dataName: __('KBps'),
    tooltipFn: () => hourlyTimeTooltip('KBps'),
    size: { height: '150px' },
    createdLabel: __('Network'),
    valueType: 'actual',
  },
  dailyPodUsageConfig: {
    chartId: 'podUsageDailyChart',
    headTitle: __('Pod Creation and Deletion Trends'),
    timeFrame: __('Last 30 days'),
    createdLabel: __('Created'),
    deletedLabel: __('Deleted'),
    tooltipFn: dailyPodTimeTooltip,
    point: { r: 1 },
    size: { height: '150px' },
    grid: { y: { show: false } },
    setAreaChart: true,
  },
  hourlyPodUsageConfig: {
    chartId: 'podUsageHourlyChart',
    headTitle: __('Pod Creation and Deletion Trends'),
    timeFrame: __('Last 24 hours'),
    createdLabel: __('Created'),
    deletedLabel: __('Deleted'),
    tooltipFn: hourlyPodTimeTooltip,
    point: { r: 1 },
    size: { height: '145px' },
    grid: { y: { show: false } },
    setAreaChart: true,
  },
  dailyImageUsageConfig: {
    chartId: 'imageUsageDailyChart',
    headTitle: __('New Image Usage Trend'),
    timeFrame: __('Last 30 days'),
    createdLabel: __('Images'),
    tooltipFn: () => dailyTimeTooltip('KBps'),
    units: __('KBps'),
    point: { r: 1 },
    size: { height: '150px' },
    grid: { y: { show: false } },
    setAreaChart: true,
  },
  hourlyImageUsageConfig: {
    chartId: 'imageUsageHourlyChart',
    headTitle: __('New Image Usage Trend'),
    timeFrame: __('Last 24 hours'),
    createdLabel: __('Images'),
    tooltipFn: () => hourlyTimeTooltip('KBps'),
    units: __('KBps'),
    point: { r: 1 },
    size: { height: '150px' },
    grid: { y: { show: false } },
    setAreaChart: true,
  },
  availableServersUsageConfig: {
    chartId: 'serverAvailabilityChart',
    title: __('Servers Available'),
    units: __('Server'),
    usageDataName: __('Used'),
    legendLeftText: __('Last 30 Days'),
    legendRightText: '',
    numDays: 30,
    availableDataName: __('Available'),
  },
  availableServersUsagePieConfig: {
    chartId: 'serverAvailablePieChart_',
  },
  serversHealthUsageConfig: {
    chartId: 'serverHealthChart',
    title: __('Servers Health'),
    units: __('Server'),
    usageDataName: __('Used'),
    legendLeftText: __('Last 30 Days'),
    legendRightText: '',
    numDays: 30,

  },
  recentServersConfig: {
    chartId: 'recentServersChart',
    tooltipFn: () => dailyTimeTooltip(),
    units: __('Servers'),
    point: { r: 1 },
    size: { height: '145px' },
    grid: { y: { show: false } },
    setAreaChart: true,
    legendLeftText: __('Last 30 Days'),
  },
  recentInstancesConfig: {
    chartId: 'recentInstancesChart',
    tooltipFn: () => dailyTimeTooltip(),
    units: __('Instances'),
    point: { r: 1 },
    size: { height: '145px' },
    grid: { y: { show: false } },
    setAreaChart: true,
    legendLeftText: __('Last 30 Days'),
  },
  recentImagesConfig: {
    chartId: 'recentInstancesChart',
    tooltipFn: () => dailyTimeTooltip(),
    units: __('Images'),
    point: { r: 1 },
    size: { height: '145px' },
    grid: { y: { show: false } },
    setAreaChart: true,
    legendLeftText: __('Last 30 Days'),
  },
};
