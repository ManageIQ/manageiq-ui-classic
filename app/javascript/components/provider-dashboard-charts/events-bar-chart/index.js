import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { http } from '../../../http_api';
import EmptyChart from "../emptyChart";
import StackBarChartGraph from "../../carbon-charts/stackBarChart";

const EventChart = ({
                     providerId, apiUrl, dataPoint1, dataPoint2, dataPointAvailable, title,
                   }) => {
  const [data, setCardData] = useState({ loading: true });

  useEffect(() => {
    const url = `/${apiUrl}/${providerId}`;

    http.get(url)
      .then((response) => {
        setCardData({
          loading: false,
          vms: response.data.aggEvents ,
        });
      });
  }, []);
  if (data.loading) return null;

  const chart_options = {
    title,
    axes: {
      left: {
        mapsTo: 'value',
        stacked: true,
      },
      bottom: {
        mapsTo: 'key',
        scaleType: 'labels',
      },
    },
    height: '400px',
    tooltip: {
      truncation: {
        type: 'none',
      },
    },
    color: {
      pairing: {
        option: 2
      },
      scale: {
        fixed: "#1A52F3",
        not_fixed: "#FF0000"
      }
    }
  };

  return (
    <div className="card-pf card-pf-utilization">
      <div className="card-pf-heading">
        <h2 className="card-pf-title">{title}</h2>
      </div>
      <div className="card-pf-body">
        {(data.vms.length > 0) ? <StackBarChartGraph data={data.vms} title={title} chart_options={chart_options} />
        : <EmptyChart />}
      </div>
    </div>
  );
};

EventChart.propTypes = {
  providerId: PropTypes.string.isRequired,
  apiUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default EventChart;
