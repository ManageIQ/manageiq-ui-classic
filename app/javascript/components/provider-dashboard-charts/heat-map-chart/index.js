import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import HeatMapChartGraph from './HeatMapChartGraph';
import { http } from '../../../http_api';

const HeatChart = ({
  providerId, apiUrl, dataPoint1, dataPoint2, dataPointAvailable, title,
}) => {
  const [data, setCardData] = useState({ loading: true });

  useEffect(() => {
    const url = `/${apiUrl}/${providerId}`;
    http.get(url)
      .then((response) => {
        setCardData({
          loading: false,
          vms: response.data.heatmaps,
        });
      });
  }, []);

  if (data.loading) return null;

  return (
    <div className="card-pf card-pf-utilization">
      <div className="card-pf-heading">
        <h2 className="card-pf-title">{title}</h2>
      </div>
      <div className="card-pf-body">
        <HeatMapChartGraph
          data={data.vms}
          dataPoint1={dataPoint1}
          dataPoint2={dataPoint2}
          dataPointAvailable={dataPointAvailable}
          title={title}
        />
      </div>
    </div>
  );
};

HeatChart.propTypes = {
  providerId: PropTypes.string.isRequired,
  apiUrl: PropTypes.string.isRequired,
  dataPoint1: PropTypes.string.isRequired,
  dataPoint2: PropTypes.string,
  dataPointAvailable: PropTypes.bool,
  title: PropTypes.string.isRequired,
};

HeatChart.defaultProps = {
  dataPointAvailable: true,
  dataPoint2: '',
};

export default HeatChart;
