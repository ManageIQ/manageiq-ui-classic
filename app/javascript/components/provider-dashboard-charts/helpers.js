import moment from 'moment';

export const getConvertedData = (data, units) => {
  if (data && data.xData && data.yData) {
    const columnsData = data.xData;
    const rowsData = data.yData;
    const arr = [];
    if (columnsData.length > 0) {
      columnsData.forEach((item, i) => {
        const obj = {};
        obj.group = units;
        obj.date = moment(item).tz(ManageIQ.timezone || 'UTC').format('MM/DD/YYYY HH:mm:ss z')
        obj.value = rowsData[i];
        arr.push(obj);
      });
    }
    return arr;
  }
  return [];
};

export const getLatestValue = (data) => {
  if (data && data.yData) {
    return data.yData[data.yData.length - 1];
  }
  return 0;
};

export const getHeatMapData = (data) => {
  if (data) {
    const arr = [];
    data.forEach((item) => {
      const obj = {};
      obj.percent = item.percent;
      obj.total = item.total;
      obj.provider = item.provider;
      obj.node = item.node;
      obj.value = item.percent * 100;
      obj.unit = item.unit;
      arr.push(obj);
    });
    return arr;
  }
  return [];
};

export const getPodsData = (data, createdlabel, deletedLabel) => {
  if (data && data.xData && data.yCreated && data.yDeleted) {
    const columnsData = data.xData;
    const createdData = data.yCreated;
    const deletedData = data.yDeleted;
    const arr = [];
    if (createdData.length > 0) {
      createdData.forEach((item, i) => {
        const obj = {};
        obj.group = createdlabel;
        obj.date = columnsData[i];
        obj.value = item;
        arr.push(obj);
      });
    }
    if (deletedData.length > 0) {
      deletedData.forEach((item, i) => {
        const obj = {};
        obj.group = deletedLabel;
        obj.date = columnsData[i];
        obj.value = item;
        arr.push(obj);
      });
    }
    return arr;
  }
  return [];
};
