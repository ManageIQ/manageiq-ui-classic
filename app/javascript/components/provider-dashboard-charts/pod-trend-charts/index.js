import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import PodsAreaChart from './podsAreaChart';
import { http } from '../../../http_api';
import { chartConfig } from '../charts_config';

const PodsTrendChart = ({
  providerId, config, apiUrl, dataPoint, toolbarEnabled,
}) => {
  const [data, setCardData] = useState({ loading: true });

  useEffect(() => {
    const url = `/${apiUrl}/${providerId}`;
    http.get(url)
      .then((response) => {
        setCardData({
          loading: false,
          vms: response.data,
        });
      });
  }, []);

  if (data.loading) return null;
  const resultData = data.vms;
  const configName = resultData[dataPoint].interval_name + config;

  return (
    <div className="card-pf card-pf-utilization">
      <div className="card-pf-heading">
        <h2 className="card-pf-title">{chartConfig[configName].headTitle}</h2>
      </div>
      <div className="card-pf-body">
        <PodsAreaChart data={data.vms} config={chartConfig[configName]} dataPoint={dataPoint} toolbarEnabled={toolbarEnabled} />
      </div>
    </div>
  );
};

PodsTrendChart.propTypes = {
  providerId: PropTypes.string.isRequired,
  config: PropTypes.string.isRequired,
  apiUrl: PropTypes.string.isRequired,
  dataPoint: PropTypes.string.isRequired,
  toolbarEnabled: PropTypes.bool,
};

PodsTrendChart.defaultProps = {
  toolbarEnabled: true,
};

export default PodsTrendChart;
