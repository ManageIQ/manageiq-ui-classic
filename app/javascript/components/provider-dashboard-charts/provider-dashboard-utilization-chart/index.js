import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import UtilizationDonutChart from './utilizationDonutChart';
import UtilizationMemoryDonutChart from './utilizationMemoryDonutChart';
import { chartConfig } from '../charts_config';
import { http } from '../../../http_api';
import EmptyChart from '../emptyChart';

const UtilizationChartGraph = ({
  providerId, title, cpuConfig, memoryConfig, dashboard,
}) => {
  const [data, setCardData] = useState({ loading: true });

  useEffect(() => {
    const url = `/${dashboard}/ems_utilization_data/${providerId}`;
    http.get(url)
      .then((response) => {
        setCardData({
          loading: false,
          metricsData: response.data.ems_utilization,
        });
      });
  }, []);

  const processMetricsData = (metrics) => {
    let a = {};
    if (metrics) {
      const data = metrics.xy_data ? metrics.xy_data : metrics;
      const keys = Object.keys(data);
      keys.forEach((item, i) => {
        if (data[item] === null) {
          a[keys[i]] = { dataAvailable: false };
          a[keys[i]].data = [];
        } else {
          a[keys[i]] = { dataAvailable: true };
          a[keys[i]].data = data[item];
        }
      });
      a.dataAvailable = true;
    } else {
      a = { dataAvailable: false };
    }
    return a;
  };

  const processData = processMetricsData(data.metricsData);

  if (data.loading) return null;
  if (!processData.dataAvailable) return <EmptyChart />;

  return (
    <div className="card-pf card-pf-utilization">
      <div className="card-pf-heading">
        <h2 className="card-pf-title">{title}</h2>
      </div>
      <div className="card-pf-body">
        <div className="row">
          <div className="col-xs-12 col-sm-12 col-md-12 col-lg-6 example-trend-container">
            {processData.cpu.dataAvailable
              ? (
                <UtilizationDonutChart
                  data={data.metricsData}
                  config={chartConfig[cpuConfig]}
                  title={title}
                />
              )
              : <EmptyChart />}
          </div>
          <div className="col-xs-12 col-sm-12 col-md-12 col-lg-6 example-trend-container">
            {processData.memory.dataAvailable
              ? (
                <UtilizationMemoryDonutChart
                  data={data.metricsData}
                  config={chartConfig[memoryConfig]}
                  title={title}
                />
              )
              : <EmptyChart />}
          </div>
        </div>
      </div>
    </div>
  );
};

UtilizationChartGraph.propTypes = {
  providerId: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  cpuConfig: PropTypes.string.isRequired,
  memoryConfig: PropTypes.string.isRequired,
  dashboard: PropTypes.string.isRequired,
};

export default UtilizationChartGraph;
