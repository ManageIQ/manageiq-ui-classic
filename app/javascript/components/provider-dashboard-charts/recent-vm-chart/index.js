import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import VmAreaChart from './vmAreaChart';
import { http } from '../../../http_api';
import { chartConfig } from '../charts_config';

const RecentVmGraph = ({
  providerId, title, config, apiUrl, dataPoint,
}) => {
  const [data, setCardData] = useState({ loading: true });

  useEffect(() => {
    const url = `/${apiUrl}/${providerId}`;
    http.get(url)
      .then((response) => {
        setCardData({
          loading: false,
          vms: response.data,
        });
      });
  }, []);

  if (data.loading) return null;
  return (
    <div className="card-pf card-pf-utilization">
      <div className="card-pf-heading">
        <h2 className="card-pf-title">{title}</h2>
      </div>
      <div className="card-pf-body">
        <VmAreaChart data={data.vms} config={chartConfig[config]} dataPoint={dataPoint} title={title} />
      </div>
    </div>
  );
};

RecentVmGraph.propTypes = {
  providerId: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  config: PropTypes.string.isRequired,
  apiUrl: PropTypes.string.isRequired,
  dataPoint: PropTypes.string.isRequired,
};

export default RecentVmGraph;
