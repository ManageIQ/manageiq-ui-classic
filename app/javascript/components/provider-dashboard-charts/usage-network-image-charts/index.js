import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import UsageAreaChart from './usageAreaChart';
import { http } from '../../../http_api';
import { chartConfig } from '../charts_config';
import EmptyChart from '../emptyChart';

const UsageTrendChart = ({
  providerId, config, apiUrl, dataPoint,
}) => {
  const [data, setCardData] = useState({ loading: true });

  useEffect(() => {
    const url = `/${apiUrl}/${providerId}`;
    http.get(url)
      .then((response) => {
        setCardData({
          loading: false,
          vms: response.data,
        });
      });
  }, []);

  if (data.loading) return null;

  const resultData = data.vms;
  const configName = resultData[dataPoint].interval_name + config;
  const title = chartConfig[configName] ? chartConfig[configName].headTitle : __('New Image Usage Trend');
  return (
    <div className="card-pf card-pf-utilization">
      <div className="card-pf-heading">
        <h2 className="card-pf-title">{title}</h2>
      </div>
      <div className="card-pf-body">
        {resultData[dataPoint].dataAvailable ? <UsageAreaChart data={data.vms} config={chartConfig[configName]} dataPoint={dataPoint} title={title} />
          : <EmptyChart /> }
      </div>
    </div>
  );
};

UsageTrendChart.propTypes = {
  providerId: PropTypes.string.isRequired,
  config: PropTypes.string.isRequired,
  apiUrl: PropTypes.string.isRequired,
  dataPoint: PropTypes.string.isRequired,
};

export default UsageTrendChart;
