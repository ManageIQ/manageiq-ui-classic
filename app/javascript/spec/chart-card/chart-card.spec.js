import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ChartCard from '../../components/container-projects/chart-card';
import { chartConfig } from '../../components/provider-dashboard-charts/charts_config';

describe('Chart Card component', () => {
  it('should render the loading card', () => {
    const wrapper = shallow(<ChartCard title="Pods" isLoading />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render the empty card', () => {
    const wrapper = shallow(<ChartCard chartType="empty" title="Pods" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render the area chart card', () => {
    const chartData = [
      { date: '2022-05-09T23:00:00.000Z', value: 0.01 },
      { date: '2022-05-10T00:00:00.000Z', value: 0.02 },
      { date: '2022-05-10T01:00:00.000Z', value: 0.02 },
      { date: '2022-05-10T02:00:00.000Z', value: 0.02 },
      { date: '2022-05-10T03:00:00.000Z', value: 0.01 },
    ];
    const getTooltipHTML = (pointData, tooltipType, unit) => {
      const date = moment(pointData[0].date);
      if (tooltipType === 'daily') {
        return (`${date.format('MM/DD/YYYY')}: ${pointData[0].value} ${unit}`);
      }
      return (`${date.format('h:mm A')}: ${pointData[0].value} ${unit}`);
    };
    const options = {
      toolbar: { enabled: false },
      tooltip: {
        customHTML: (pointData) => getTooltipHTML(pointData, 'hourly', __('Cores')),
        truncation: {
          type: 'none',
        },
      },
      height: '50px',
      grid: {
        x: {
          enabled: false,
        },
        y: {
          enabled: false,
        },
      },
      axes: {
        bottom: {
          visible: false,
          mapsTo: 'date',
          scaleType: 'time',
        },
        left: {
          visible: false,
          mapsTo: 'value',
          scaleType: 'linear',
        },
      },
      points: {
        enabled: false,
      },
      legend: {
        enabled: false,
      },
    };
    const wrapper = shallow(<ChartCard
      title="CPU Utilization"
      textNumber={0.02}
      textUnit="Cores"
      textTitle="CPU Utilization"
      chartData={chartData}
      options={options}
      isLoading={false}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render the quotas chart card', () => {
    const chartData = [
      {
        resource: 'cpu', quota_enforced: 20, quota_observed: '0.3', units: 'Cores',
      },
      {
        resource: 'memory', quota_enforced: 1, quota_observed: 0, units: 'GB',
      },
      {
        resource: 'pods', quota_enforced: 10, quota_observed: 4, units: '',
      },
      {
        resource: 'replicationcontrollers', quota_enforced: 5, quota_observed: 7, units: '',
      },
      {
        resource: 'resourcequotas', quota_enforced: 1, quota_observed: 1, units: '',
      },
      {
        resource: 'services', quota_enforced: 5, quota_observed: 4, units: '',
      },
    ];
    const wrapper = shallow(<ChartCard chartType="quotas" title="Quotas" chartData={chartData} isLoading={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render the pods trend chart card', () => {
    const chartData = {
      pod_metrics: {
        dataAvailable: true,
        interval_name: 'hourly',
        xy_data: {
          dataAvailable: true,
          xData: [
            '2022-05-09T23:00:00.000Z',
            '2022-05-10T00:00:00.000Z',
            '2022-05-10T01:00:00.000Z',
          ],
          yCreated: [1, 0, 0],
          yDeleted: [0, 0, 1],
        },
      },
    };
    const wrapper = shallow(<ChartCard
      chartType="podsTrend"
      title="Pod Creation and Deletion Trends"
      chartData={chartData}
      isLoading={false}
      config={chartConfig.hourlyPodUsageConfig}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render the pods table card', () => {
    const chartData = {
      headers: [
        { key: 'name', header: __('Name') },
        { key: 'phase', header: __('Status') },
        { key: 'ready_condition_status', header: __('Ready Status') },
        { key: 'running_containers_summary', header: __('Ready Containers') },
      ],
      rows: [
        {
          name: 'test0',
          phase: 'Succeeded',
          running_containers_summary: '0/1',
          ready_condition_status: 'False',
          id: '0.03272110141753015',
        },
        {
          name: 'test1',
          phase: 'Running',
          running_containers_summary: '1/1',
          ready_condition_status: 'True',
          id: '0.548571332866931',
        },
        {
          name: 'test2',
          phase: 'Running',
          running_containers_summary: '1/1',
          ready_condition_status: 'True',
          id: '0.988113698073033',
        },
      ],
    };
    const wrapper = shallow(<ChartCard chartType="table" title="Pods" chartData={chartData} isLoading={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
