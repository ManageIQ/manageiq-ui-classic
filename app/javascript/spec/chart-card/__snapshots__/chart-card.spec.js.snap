// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Card component should render the area chart card 1`] = `
<AreaChartCard
  chartData={
    Array [
      Object {
        "date": "2022-05-09T23:00:00.000Z",
        "value": 0.01,
      },
      Object {
        "date": "2022-05-10T00:00:00.000Z",
        "value": 0.02,
      },
      Object {
        "date": "2022-05-10T01:00:00.000Z",
        "value": 0.02,
      },
      Object {
        "date": "2022-05-10T02:00:00.000Z",
        "value": 0.02,
      },
      Object {
        "date": "2022-05-10T03:00:00.000Z",
        "value": 0.01,
      },
    ]
  }
  options={
    Object {
      "axes": Object {
        "bottom": Object {
          "mapsTo": "date",
          "scaleType": "time",
          "visible": false,
        },
        "left": Object {
          "mapsTo": "value",
          "scaleType": "linear",
          "visible": false,
        },
      },
      "grid": Object {
        "x": Object {
          "enabled": false,
        },
        "y": Object {
          "enabled": false,
        },
      },
      "height": "50px",
      "legend": Object {
        "enabled": false,
      },
      "points": Object {
        "enabled": false,
      },
      "toolbar": Object {
        "enabled": false,
      },
      "tooltip": Object {
        "customHTML": [Function],
        "truncation": Object {
          "type": "none",
        },
      },
    }
  }
  textNumber={0.02}
  textTitle="CPU Utilization"
  textUnit="Cores"
  title="CPU Utilization"
/>
`;

exports[`Chart Card component should render the empty card 1`] = `
<EmptyCard
  title="Pods"
/>
`;

exports[`Chart Card component should render the loading card 1`] = `
<div
  className="card-pf card-pf-aggregate-status card-pf-accented chart-card"
>
  <h2
    className="card-title"
  >
    Pods
  </h2>
  <div
    className="card-pf-body"
  >
    <Loading
      active={true}
      className="export-spinner"
      description="Active loading indicator"
      small={true}
      withOverlay={false}
    />
  </div>
</div>
`;

exports[`Chart Card component should render the pods table card 1`] = `
<TableCard
  chartData={
    Object {
      "headers": Array [
        Object {
          "header": "Name",
          "key": "name",
        },
        Object {
          "header": "Status",
          "key": "phase",
        },
        Object {
          "header": "Ready Status",
          "key": "ready_condition_status",
        },
        Object {
          "header": "Ready Containers",
          "key": "running_containers_summary",
        },
      ],
      "rows": Array [
        Object {
          "id": "0.03272110141753015",
          "name": "test0",
          "phase": "Succeeded",
          "ready_condition_status": "False",
          "running_containers_summary": "0/1",
        },
        Object {
          "id": "0.548571332866931",
          "name": "test1",
          "phase": "Running",
          "ready_condition_status": "True",
          "running_containers_summary": "1/1",
        },
        Object {
          "id": "0.988113698073033",
          "name": "test2",
          "phase": "Running",
          "ready_condition_status": "True",
          "running_containers_summary": "1/1",
        },
      ],
    }
  }
  title="Pods"
/>
`;

exports[`Chart Card component should render the pods trend chart card 1`] = `
<TrendChartCard
  chartData={
    Object {
      "pod_metrics": Object {
        "dataAvailable": true,
        "interval_name": "hourly",
        "xy_data": Object {
          "dataAvailable": true,
          "xData": Array [
            "2022-05-09T23:00:00.000Z",
            "2022-05-10T00:00:00.000Z",
            "2022-05-10T01:00:00.000Z",
          ],
          "yCreated": Array [
            1,
            0,
            0,
          ],
          "yDeleted": Array [
            0,
            0,
            1,
          ],
        },
      },
    }
  }
  config={
    Object {
      "chartId": "podUsageHourlyChart",
      "createdLabel": "Created",
      "deletedLabel": "Deleted",
      "grid": Object {
        "y": Object {
          "show": false,
        },
      },
      "headTitle": "Pod Creation and Deletion Trends",
      "point": Object {
        "r": 1,
      },
      "setAreaChart": true,
      "size": Object {
        "height": "145px",
      },
      "timeFrame": "Last 24 hours",
      "tooltipFn": [Function],
    }
  }
  title="Pod Creation and Deletion Trends"
/>
`;

exports[`Chart Card component should render the quotas chart card 1`] = `
<BarChartCard
  chartData={
    Array [
      <div
        className="quota-chart"
      >
        <e
          data={
            Array [
              Object {
                "group": "cpu - 0.3 Cores Used",
                "value": 1.5,
              },
            ]
          }
          options={
            Object {
              "height": "75px",
              "meter": Object {
                "status": Object {
                  "ranges": Array [
                    Object {
                      "range": Array [
                        0,
                        20,
                      ],
                    },
                  ],
                },
              },
              "toolbar": Object {
                "enabled": false,
              },
            }
          }
        />
      </div>,
      <div
        className="quota-chart"
      >
        <e
          data={
            Array [
              Object {
                "group": "memory - 0 GB Used",
                "value": 0,
              },
            ]
          }
          options={
            Object {
              "height": "75px",
              "meter": Object {
                "status": Object {
                  "ranges": Array [
                    Object {
                      "range": Array [
                        0,
                        1,
                      ],
                    },
                  ],
                },
              },
              "toolbar": Object {
                "enabled": false,
              },
            }
          }
        />
      </div>,
      <div
        className="quota-chart"
      >
        <e
          data={
            Array [
              Object {
                "group": "pods - 4  Used",
                "value": 40,
              },
            ]
          }
          options={
            Object {
              "height": "75px",
              "meter": Object {
                "status": Object {
                  "ranges": Array [
                    Object {
                      "range": Array [
                        0,
                        10,
                      ],
                    },
                  ],
                },
              },
              "toolbar": Object {
                "enabled": false,
              },
            }
          }
        />
      </div>,
      <div
        className="quota-chart"
      >
        <e
          data={
            Array [
              Object {
                "group": "replicationcontrollers - 7  Used",
                "value": 140,
              },
            ]
          }
          options={
            Object {
              "height": "75px",
              "meter": Object {
                "status": Object {
                  "ranges": Array [
                    Object {
                      "range": Array [
                        0,
                        5,
                      ],
                    },
                  ],
                },
              },
              "toolbar": Object {
                "enabled": false,
              },
            }
          }
        />
      </div>,
      <div
        className="quota-chart"
      >
        <e
          data={
            Array [
              Object {
                "group": "resourcequotas - 1  Used",
                "value": 100,
              },
            ]
          }
          options={
            Object {
              "height": "75px",
              "meter": Object {
                "status": Object {
                  "ranges": Array [
                    Object {
                      "range": Array [
                        0,
                        1,
                      ],
                    },
                  ],
                },
              },
              "toolbar": Object {
                "enabled": false,
              },
            }
          }
        />
      </div>,
      <div
        className="quota-chart"
      >
        <e
          data={
            Array [
              Object {
                "group": "services - 4  Used",
                "value": 80,
              },
            ]
          }
          options={
            Object {
              "height": "75px",
              "meter": Object {
                "status": Object {
                  "ranges": Array [
                    Object {
                      "range": Array [
                        0,
                        5,
                      ],
                    },
                  ],
                },
              },
              "toolbar": Object {
                "enabled": false,
              },
            }
          }
        />
      </div>,
    ]
  }
  title="Quotas"
/>
`;
